### PR TITLE
Update arch diagram to remove WebGL

### PR DESCRIPTION
**Pull request by dontcallmedom**
also provide SVG source

### CODE REVIEW

**Review by anssiko — APPROVED** [review]
Nice work, thanks!

TIL: [Inkscape can convert PNG to SVG](https://linuxhint.com/convert-png-to-svg-inkscape/).

**Comment by anssiko** [comment]
One detail to note: some frameworks such as TF.js still use a [WebGL backend](https://www.tensorflow.org/js/guide/platform_environment#webgl_backend). I think this is illustrated by the arrow pointing downwards toward the WebGL/WebGPU box. For conciseness and future-proofing reasons I'm supportive of removing WebGL from this diagram given WebNN dropped support and is the context here.

Given this is now in an SVG format we can easily maintain this diagram as our architecture evolves.

**Comment by huningxin** [comment]
For the change on macOS/iOS, the platform API is changed from MLCompute to BNNS/MPS, is that suggested by Apple folks?

**Comment by dontcallmedom** [comment]
> For the change on macOS/iOS, the platform API is changed from MLCompute to BNNS/MPS, is that suggested by Apple folks?

Not at all - I'm not sure how I came to have that change in my SVG version of the diagram; I corrected that to MLCompute.

File: content/webnn_arch.svg
Comment (huningxin) [inline content/webnn_arch.svg:1]: webnn_arch.svg LGTM.

File: content/webnn_arch.png
Comment (huningxin) [inline content/webnn_arch.png:1]: You may miss updating content/webnn_arch.png. It still changes ML Compute to BNNS/MPS.

**Review by huningxin — APPROVED** [review]
LGTM, thanks @dontcallmedom !

**Pull request merged**